policy/scan/local_scanner.go

```diff
@@ -159,6 +162,14 @@ func (s *LocalScanner) distributeJob(job *Job, ctx context.Context, upstreamConf
 		if err != nil {
 			return nil, false, err
 		}
+
+		// TODO: this is a temporary fix for populating the platform information for assets that only set it
+		// during a scan, e.g. container images. This should be removed once we start using MQL for retrieving
+		// platform information.
+		if err := setPlatforms(ctx, assetList, im, job.DoRecord); err != nil {
```

Comment:
this is problematic. Opening a connection can be very expensive. I think it would be better if you opened the connection, get the platform id, call sync asset, then scan the asset.

I think ebs scanning is one of the things that will suffer from this. cc @vjeffrey

Reply:
I could do that. The only downside is that there will be 2 code paths that do asset synchronization

Reply:
i'm not sure i understand why there would be 2

Reply:
I changed it. Not sure if that is exactly what you had in mind

Reply:
what is the original problem this is all fixing? is it that some assets do not get platform information until they are scanned?

Reply:
@vjeffrey yup, previously we had a `SyncAssets` call after the scan was done that was populating that info (pre-v7). Thats no longer there

Reply:
i think i'm still unclear, bc there also other cases where we don't have the platform info, right? e.g. aws with ec2 discovery, we likely won't be able to connect directly to all the instances, so a bunch of them won't have the platform info unless we scan the instance directly.

Reply:
This is not the latest change. In the latest change the assets which platform changes are being sync-ed

Reply:
👍  got it, so basically anything that doesn't get platform information during the first sync call never gets its platform information in the console, because we're no longer calling sync/set asset as part of the scan path. 

just tested this with ebs volume scanning, it definitely suffers from the same issue. a lot of stuff is going to suffer from this (ec2 instances that are discovered along with an aws account, for example)